toolchain/lower/mangler.cpp

```diff
@@ -0,0 +1,99 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/lower/mangler.h"
+
+#include "toolchain/base/kind_switch.h"
+#include "toolchain/sem_ir/entry_point.h"
+
+namespace Carbon::Lower {
+
+auto Mangler::MangleInverseQualifiedNameScope(llvm::raw_ostream& os,
+                                              SemIR::NameScopeId name_scope_id)
+    -> void {
+  struct NameEntry {
+    SemIR::NameScopeId name_id;
+    char prefix;
```

Comment:
Can you document the intended semantics of `prefix`? It looks like `:`, `.`, and `\0` are intended values here -- using `\0` in particular seems very subtle since you seem to be relying on implicit 0-init for that.

Reply:
Ah, sure - documented the 3 cases (& also removed dependence on implicit 0-init, FWIW) in the comment. Though not sure if it'd be better to document the cases at their usage, and leave the comment here generically describing any prefix and the null-means-no-prefix case.